zcash_primitives/src/transaction/builder.rs

```diff
@@ -1575,24 +1556,19 @@ mod tests {
                     recipient: address,
                     value: NoteValue::from_raw(42),
                 }),
+                false,
             )
             .unwrap();
         builder
-            .add_recipient::<FeeError>(&asset_desc_2, address, NoteValue::from_raw(21))
+            .add_recipient::<FeeError>(&asset_desc_2, address, NoteValue::from_raw(21), false)
```

Comment:
That is very good but we need one test with `first_issue = true` and see that we indeed get the referance note as the first note of the action (check correct number of notes + check that the **first** note of the action is really a ref note.)

Reply:
Added few tests